database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -255,6 +252,9 @@ public void onDataChange(DataSnapshot snapshot) {
                     Log.w(TAG, "Key not found at ref: " + snapshot.getRef());
                 }
             }
+
+            mKeysWithPendingUpdate.remove(key);
```

Comment:
I'm glad you slowed me down on this one because I didn't notice I was introducing a bug where `onDataChanged` would be called twice in quick succession. Everything LGTM now.

PS: I didn't want to make the code explanation too verbose, but basically if this line is reached:
```java
Log.w(TAG, "Key not found at ref: " + snapshot.getRef());
```
then the bug occurs because we won't pop the queue.